src/matrix/registration/Registration.ts

```diff
@@ -108,6 +109,8 @@ export class Registration {
                 return new DummyAuth(session, params?.[type]);
             case "m.login.terms":
                 return new TermsAuth(session, params?.[type]);
+            case TOKEN_AUTH_TYPE:
```

Comment:
accept both the prefixed and non-prefixed type here, pass in the actual type used by the server and use that in generateAuthenticationData, but perhaps return the non-prefixed type to return from `get type` so it's easy to compare when seeing which stage you're in...